src/DotLiquid/Template.cs

```diff
@@ -58,6 +63,7 @@ static Template()
             RegexTimeOut = TimeSpan.FromSeconds(10);
             NamingConvention = new RubyNamingConvention();
             FileSystem = new BlankFileSystem();
+            SyntaxCompatibilityLevelDefault = SyntaxCompatibility.DotLiquid20;

```

Comment:
I'd love to know the use case that you cant use the default and cant use the more flexible explicit RenderParameters. Having optional attributes become unwieldly as they are added to. If we change all options from static to the same 'default' behavior we would have a mess and implemented this we would have a mess. Im trying to think ahead.

Reply:
See https://github.com/dotliquid/dotliquid/wiki/DotLiquid-Syntax-Compatibility#strict-liquid-syntax-work-in-progress for the no longer theoretical implementation of setting the syntax in a template.

Reply:
We should be able to use the "public string Render(RenderParameters parameters)" overload by passing all the render parameters explicitly. Thought it would be easy for the users of "public string Render(Hash localVariables, IFormatProvider formatProvider = null)" to pass in the new syntax compatibility setting instead of creating caller creating render parameters. But Agree on your point that it might become a mess when we change more options.